rpc/client/client_test.go

```diff
@@ -757,3 +757,62 @@ func TestMempool2Nodes(t *testing.T) {
 
 	assert.Equal(node2.Mempool.TxsBytes(), int64(len("good")))
 }
+
+func TestAggregationLoop(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	mockApp := &mocks.Application{}
+	mockApp.On("InitChain", mock.Anything).Return(abci.ResponseInitChain{})
+	key, _, _ := crypto.GenerateEd25519Key(crand.Reader)
+	node, err := node.NewNode(context.Background(), config.NodeConfig{
+		DALayer:    "mock",
+		Aggregator: true,
+		BlockManagerConfig: config.BlockManagerConfig{
+			BlockTime: 200 * time.Millisecond,
+		}},
+		key, proxy.NewLocalClientCreator(mockApp),
+		&tmtypes.GenesisDoc{
+			ChainID:       "test",
+			InitialHeight: 100,
+			GenesisTime:   time.Now().Local().Add(time.Second * time.Duration(100)),
+		},
+		log.TestingLogger())
+	require.NoError(err)
+	require.NotNil(node)
+
+	rpc := NewClient(node)
+	require.NotNil(rpc)
+	mockApp.On("BeginBlock", mock.Anything).Return(abci.ResponseBeginBlock{})
+	mockApp.On("EndBlock", mock.Anything).Return(abci.ResponseEndBlock{})
+	mockApp.On("Commit", mock.Anything).Return(abci.ResponseCommit{})
+	const evType = "ethereum_tx"
+	const evKey = "ethereumTxHash"
+	const evValue = "RLP encoded Keccak hash"
+	mockApp.On("DeliverTx", mock.Anything).Return(abci.ResponseDeliverTx{
```

Comment:
Instead of executing a lot of code not related to the issue tested, you can use `BeginBlock` or `EndBlock` mock, to assert that time of the call is actually after time specified in genesis. This will simplify the test a lot, and you should be able to remove the sleep part.